conformance/tests/gateway-observed-generation-bump.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, GatewayObservedGenerationBump)
+}
+
+var GatewayObservedGenerationBump = suite.ConformanceTest{
+	ShortName:   "GatewayObservedGenerationBump",
+	Description: "A Gateway in the gateway-conformance-infra namespace should update the observedGeneration in all of it's Status.Conditions after an update to the spec",
+	Manifests:   []string{"tests/gateway-observed-generation-bump.yaml"},
+	Test: func(t *testing.T, s *suite.ConformanceTestSuite) {
+
+		gwNN := types.NamespacedName{Name: "observed-generation-bump", Namespace: "gateway-conformance-infra"}
+
+		t.Run("observedGeneration should increment", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+			defer cancel()
+
+			namespaces := []string{"gateway-conformance-infra"}
+			kubernetes.NamespacesMustBeAccepted(t, s.Client, s.TimeoutConfig, namespaces)
+
+			existing := &v1beta1.Gateway{}
+			err := s.Client.Get(ctx, gwNN, existing)
+			require.NoErrorf(t, err, "error getting Gateway: %v", err)
+
+			// Sanity check
+			if kubernetes.GatewayConditionsHaveLatestObservedGeneration(existing) {
```

Comment:
shouldn't this be as below? (here and in all other similar places)
```suggestion
			if !kubernetes.GatewayConditionsHaveLatestObservedGeneration(existing) {
```